specs/core/1_custody-game.md

```diff
@@ -336,37 +172,27 @@ def legendre_bit(a: int, q: int) -> int:
         return 0
 ```
 
-### `custody_subchunkify`
+### `custody_atoms`
 
-Given one proof of custody chunk, returns the proof of custody subchunks of the correct sizes.
+Given one set of data, return the custody atoms: each atom will be combined with one legendre bit.
 
 ```python
-def custody_subchunkify(bytez: bytes) -> Sequence[bytes]:
-    bytez += b'\x00' * (-len(bytez) % BYTES_PER_CUSTODY_SUBCHUNK)
-    return [bytez[i:i + BYTES_PER_CUSTODY_SUBCHUNK]
-            for i in range(0, len(bytez), BYTES_PER_CUSTODY_SUBCHUNK)]
+def get_custody_atoms(bytez: bytes) -> Sequence[bytes]:
+    bytez += b'\x00' * (-len(bytez) % BYTES_PER_CUSTODY_ATOM)  # right-padding
+    return [bytez[i:i + BYTES_PER_CUSTODY_ATOM]
+            for i in range(0, len(bytez), BYTES_PER_CUSTODY_ATOM)]
 ```
 
-### `get_custody_chunk_bit`
+### `compute_custody_bit`
 
 ```python
-def get_custody_chunk_bit(key: BLSSignature, chunk: bytes) -> bool:
+def compute_custody_bit(key: BLSSignature, data: bytes) -> bit:
     full_G2_element = bls_signature_to_G2(key)
     s = full_G2_element[0].coeffs
-    bits = [legendre_bit((i + 1) * s[i % 2] + int.from_bytes(subchunk, "little"), BLS12_381_Q)
-            for i, subchunk in enumerate(custody_subchunkify(chunk))]
-
-    return bool(sum(bits) % 2)
-```
-
-### `get_chunk_bits_root`
-
-```python
-def get_chunk_bits_root(chunk_bits: Bitlist[MAX_CUSTODY_CHUNKS]) -> bit:
-    aggregated_bits = 0
-    for i, b in enumerate(chunk_bits):
-        aggregated_bits += 2**i * b
-    return legendre_bit(aggregated_bits, BLS12_381_Q)
+    bits = [legendre_bit((i + 1) * s[i % 2] + int.from_bytes(atom, "little"), BLS12_381_Q)
```

Comment:
This is the old construction, broken by Dmitry. It needs to be replaced by legendre_bit(sum(s[i % 2]**i * int.from_bytes(atom, "little")), BLS12_381_Q)